File: pirulito2.R
library(dplyr)
library(ggplot2)
library(hrbrthemes)
library(readr)

# Substitua "caminho/para/seu/arquivo.csv" pelo caminho do arquivo CSV em sua máquina
# data <- read.csv("caminho/para/seu/arquivo.csv", header = TRUE)

# Lendo os dados de um arquivo CSV hospedado em um repositório do GitHub
 data <- read.table("https://raw.githubusercontent.com/holtzy/data_to_viz/master/Example_dataset/7_OneCatOneNum.csv", header=TRUE, sep=",")

# Criar o gráfico de pirulito
  data %>%
    filter(!is.na(Value)) %>%
    arrange(Value) %>%
    tail(20) %>%
    mutate(Country=factor(Country, Country)) %>%
    ggplot( aes(x=Country, y=Value) ) +
    geom_segment( aes(x=Country ,xend=Country, y=0, yend=Value), color="grey") +
    geom_point(size=3, color="#69b3a2") +
    theme(
      panel.grid.minor.y = element_blank(),
      panel.grid.major.y = element_blank(),
      legend.position="none",
      plot.title = element_text(hjust = 0.5, size = 15)) +
    ggtitle("Quantidade de armas exportadas pelos 20 maiores exportadores em 2017") +
    xlab("países") +
    ylab("quantidade de armas") +
    coord_flip() +
    theme_ipsum() 

File: pirulito2.py
import pandas as pd
import matplotlib.pyplot as plt

# Substitua "caminho/para/seu/arquivo.csv" pelo caminho do arquivo CSV em sua máquina
# data = pd.read_csv("caminho/para/seu/arquivo.csv")

# Lê os dados do arquivo CSV fornecido pela URL e os carrega em um DataFrame chamado 'data'
data = pd.read_csv("https://raw.githubusercontent.com/holtzy/data_to_viz/master/Example_dataset/7_OneCatOneNum.csv")

# Remove as linhas com valores ausentes na coluna 'Value' e ordena o DataFrame pelos valores de 'Value' 
# em ordem ascendente, em seguida, seleciona as 20 maiores linhas
data_filtered = data.dropna(subset=['Value']).sort_values('Value').tail(20)

# Cria o gráfico de pirulito
plt.figure(figsize=(10, 6))
plt.hlines(y=data_filtered['Country'], xmin=0, xmax=data_filtered['Value'], color='grey')
plt.plot(data_filtered['Value'], data_filtered['Country'], 'o', color='#69b3a2', markersize=8)
plt.title("Quantidade de armas exportadas pelos 20 maiores exportadores em 2017")
plt.xlabel("Quantidade de armas")
plt.ylabel("Países")
plt.grid(True, axis='y', linestyle='--', alpha=0.7)
plt.tight_layout()
plt.show()
